privacyidea/api/periodictask.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+#  2018-07-09 Friedrich Weber <friedrich.weber@netknights.it>
+#             Initial implementation of periodic tasks
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+__doc__ = """These endpoints are used to create, modify and delete
+periodic tasks.
+
+This module is tested in tests/test_api_periodictask.py"""
+
+import json
+import logging
+
+from flask import Blueprint, g, request
+
+from privacyidea.api.lib.prepolicy import prepolicy, check_base_action
+from privacyidea.api.lib.utils import send_result, getParam
+from privacyidea.lib.error import ParameterError
+from privacyidea.lib.policy import ACTION
+from privacyidea.lib.log import log_with
+from privacyidea.lib.periodictask import get_periodic_tasks, set_periodic_task, delete_periodic_task, \
+    enable_periodic_task, get_periodic_task_by_id, get_taskmodule, get_available_taskmodules
+from privacyidea.lib.utils import is_true
+
+log = logging.getLogger(__name__)
+
+periodictask_blueprint = Blueprint('periodictask_blueprint', __name__)
+
+
+def convert_datetimes_to_string(ptask):
+    """
+    Convert the ``last_runs`` timestamps to ISO 8601 strings and return a copy of ``ptask``.
+    :param ptask: periodic task dictionary
+    :return: a new periodic task dictionary
+    """
```

Comment:
Wouldn't the ptask class actually be a better location for that? Like adding a method:

    ptask.get_with_date_as_string()

or some better name?

Reply:
I changed this to use AUTH_DATE_FORMAT.